GeoMesa_Feature_List.md

```diff
@@ -0,0 +1,100 @@
+  - Hooks for updating GeoServer on stream update
+    - Pointer: Implemented? [pomfile](https://github.com/locationtech/geomesa/tree/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-gs-plugin/geomesa-stream-gs-plugin)
+
+## Data Processing
+
+- geomesa-compute
+  - Generating `RDD`s of `SimpleFeature`s
+      - Pointer: [GeoMesaSpark.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala)
+      - Capable of querying with CQL to fill an RDD with some subset of your data
+    - Carrying out spark SQL queries to process geomesa data
+      - Pointer: [GeoMesaSparkSql.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/sql/GeoMesaSparkSql.scala)
+      - When constructing a spark context, "yarn-client" is set to be the master, which isn't always a good assumption
+      - As of 7/12/16, some stubbed out functions remain in the [`GeoMesaDataContext`](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/sql/GeoMesaDataContext.scala)
+- geomesa-jobs
+  - Reading data for use in a custom M/R job
+      - Pointer: [geomesa-jobs mapreduce](https://github.com/locationtech/geomesa/blob/master/geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/)
+      - Pointer: [geomesa-jobs mapred](https://github.com/locationtech/geomesa/blob/master/geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/)
+      - Apparently capable of reading from any GeoMesa `DataStore` as well as from the filesystem with or without avro files specifying the details of the conversion.
+- geomesa-process - (On Accumulo backed GeoMesa instances only - with the possible exception of `Point2Point` and `DensityProcess`, based on file locations and accumulo imports within said files. All processes are registered in https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/ProcessFactory.scala)
+  - computing a heatmap from a provided CQL query
+    - Pointer: [DensityProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/DensityProcess.scala)
+  - Given CQL and a description of the stats of interest, compute said stats on said CQL results
+    - Pointer: [StatsIteratorProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala)
```

Comment:
stats we currently support: count, enumeration, frequency (countMinSketch), histogram, topk, min/max (bounds).
we also support count, bounds, histogram and top-k through the command line tools